sdk/core/core-lro/docs/LROEngine.md

```diff
@@ -0,0 +1,158 @@
+
+Currently `options` have `intervalInMs` to control the polling interval, `resumeFrom` to enable resuming from a serialized state, and `lroResourceLocationConfig` which could determine where to find the results of the LRO after the operation is finished. Typically, Autorest figures out the value for `LroResourceLocationConfig` from the `x-ms-long-running-operation-options` swagger extension. If there are new arguments to be added to the class, they could be added to the options type.
+
+### [`LroImpl`](https://github.com/deyaaeldeen/autorest.typescript/blob/lro-simplify/src/lroImpl.ts)
+
+This class implements the `LongRunningOperation<T>` interface and I propose to make it auto-generated by Autorest. `LroImpl` needs access to a few pieces: operation specification and operation arguments and a primitive function that can take them as input to send a request and converts the received response into one of type `LroResponse<T>` which has both the flattened and the raw responses.
+
+## Usage examples
+
+### Create an object of `LroImpl`
+
+```ts
+const directSendOperation = async (
+  args: OperationArguments,
+  spec: OperationSpec
+): Promise<unknown> => {
+  return this.client.sendOperationRequest(args, spec);
+};
+const sendOperation = async (
+  args: OperationArguments,
+  spec: OperationSpec
+) => {
+  let currentRawResponse: FullOperationResponse | undefined = undefined;
+  const providedCallback = args.options?.onResponse;
+  const callback: RawResponseCallback = (
+    rawResponse: FullOperationResponse,
+    flatResponse: unknown
+  ) => {
+    currentRawResponse = rawResponse;
+    providedCallback?.(rawResponse, flatResponse);
+  };
+  const updatedArgs = {
+    ...args,
+    options: {
+      ...args.options,
+      onResponse: callback
+    }
+  };
+  const flatResponse = await directSendOperation(updatedArgs, spec);
+  return {
+    flatResponse,
+    rawResponse: {
+      statusCode: currentRawResponse!.status,
+      body: currentRawResponse!.parsedBody,
+      headers: currentRawResponse!.headers.toJSON()
+    }
+  };
+};
+
+const lro = new LroImpl(
+  sendOperation,
+  { options }, // arguments are just the operation options
+  spec
+);
+```
+
+### Using `LroEngine`
+
+```ts
+const pollerEngine = new LroEngine(lro, { intervalInMs: 2000 }); // lro was instantiated in the previous section
+const result = pollerEngine.pollUntilDone();
+```
+
+## Testing
+
+We have [extensive test suite for LROs](https://github.com/Azure/autorest.typescript/blob/main/test/integration/lro.spec.ts) in the TypeScript code generator repo. I both added those tests here and re-implemented the [lro routes](https://github.com/Azure/autorest.testserver/blob/main/legacy/routes/lros.js) in the Autorest test server. For this to work, I created a [fairly low-level instantiation for `LongRunningOperation<T>` with just `@azure/core-rest-pipeline`](https://github.com/deyaaeldeen/azure-sdk-for-js/blob/lro-design/sdk/core/core-lro/test/utils/coreRestPipelineLro.ts).
```

Comment:
@deyaaeldeen This link is broken. This causes release CI to be broken. Could you fix them? 

Also 2 links in this page is pointing to your fork instead of Azure fork. Is that expected?